packages/beacon-node/src/network/network.ts

```diff
@@ -618,11 +618,9 @@ export class Network implements INetwork {
     request: Req
   ): AsyncIterable<ResponseIncoming> {
     const fork = this.config.getForkName(this.clock.currentSlot);
-    const requestType = requestSszTypeByMethod(fork, this.config)[method];
-    const requestData = requestType ? requestType.serialize(request as never) : new Uint8Array();
```

Comment:
This means we don't serialize before sending to the network thread but we need to run `libp2p.dialProtocol` to be able to figure out which protocol to use (if there are multiple) before we can serialize the data... not sure right now how to solve this.